hudi-sync/hudi-hive-sync/src/main/java/org/apache/hudi/hive/ddl/HMSDDLExecutor.java

```diff
@@ -225,8 +225,9 @@ public void updatePartitionsToTable(String tableName, List<String> changedPartit
         String fullPartitionPath = StorageSchemes.HDFS.getScheme().equals(partitionScheme)
             ? FSUtils.getDFSFullPartitionPath(syncConfig.getHadoopFileSystem(), partitionPath) : partitionPath.toString();
         List<String> partitionValues = partitionValueExtractor.extractPartitionValuesInPath(partition);
-        sd.setLocation(fullPartitionPath);
-        return new Partition(partitionValues, databaseName, tableName, 0, 0, sd, null);
+        StorageDescriptor partitionSd = sd.deepCopy();
+        partitionSd.setLocation(fullPartitionPath);
```

Comment:
Why deep copy fixed the problem ? Modify the sd directly lost something ?

Reply:
It uses the same one storage descriptor in a loop and returns a partition containing sd's ref.

So if update multiple partitions, the sd of all partitions will be the same.

Reply:
Got you, let's make this into release 0.12.0.

Reply:
You mean the location for the partition is wrong ? What is the effect of that ?

Reply:
Yes, the location for partition is wrong. It will cause the query result to be wrong when using hive or trino. These queries rely on the partition location recorded in the metastore.

Reply:
Great catch! Let's take it in 0.12.0. We should cover adding multiple partitions in a test. I can add it later.